projects/models.py

```diff
@@ -83,34 +85,66 @@ def create_from_bibtex(self, bibtex_entry, project, username):
         pub = Publication()
 
         pub.project_id = project.id
-        if "booktitle" in bibtex_entry:
-            pub.booktitle = bibtex_entry.get("booktitle")
-        if "journal" in bibtex_entry:
-            pub.journal = bibtex_entry.get("journal")
-        if "publisher" in bibtex_entry:
-            pub.publisher = bibtex_entry.get("publisher")
+
+        pub.publication_type = bibtex_entry.get("ENTRYTYPE")
         pub.title = bibtex_entry.get("title")
         pub.year = bibtex_entry.get("year")
+        month = bibtex_entry.get("month")
         pub.author = bibtex_entry.get("author")
-        if bibtex_entry.get("abstract"):
-            pub.abstract = bibtex_entry.get("abstract")
         pub.bibtex_source = json.dumps(bibtex_entry)
         pub.added_by_username = username
+
+        pub.month = None
+        try:
+            pub.month = int(month)
+        except Exception:
+            pass
+        try:
+            pub.month = datetime.datetime.strptime(month, "%b").month
+        except Exception:
+            pass
+        try:
+            pub.month = datetime.datetime.strptime(month, "%B").month
+        except Exception:
+            pass
+
+        forum = []
+        if "journal" in bibtex_entry:
+            forum.append(bibtex_entry["journal"])
+        if "booktitle" in bibtex_entry:
+            forum.append(bibtex_entry["booktitle"])
+        if "series" in bibtex_entry:
+            forum.append(bibtex_entry["series"])
+        if "publisher" in bibtex_entry:
+            forum.append(bibtex_entry["publisher"])
+        if "school" in bibtex_entry:
+            forum.append(bibtex_entry["school"])
+        if "institution" in bibtex_entry:
+            forum.append(bibtex_entry["institution"])
+        if "address" in bibtex_entry:
+            forum.append(bibtex_entry["address"])
+        pub.forum = ",".join(forum)
```

Comment:
@diurnalist we have a bibtex source field that stores the original bibtex entry as a json field. The forum field benefits to generate this page https://chameleoncloud.org/about/chameleon-used-research/, so that we can have "In <forum>".